src/integration-tests/integration-test.py

```diff
@@ -291,301 +267,345 @@ def launch(self, path: str) -> None:
                 continue
 
-            self.logger.info("launching repro: %s", self.target_jobs[job_id])
-            report = self.get_job_crash(job_id)
+            report = self.get_job_crash_report(job.job_id)
             if report is None:
-                self.logger.warning(
-                    "target does not include crash reports: %s",
-                    self.target_jobs[job_id],
+                self.logger.error(
+                    "target does not include crash reports: %s", job.config.name
                 )
-                return
-            (container, path) = report
-            self.repros[job_id] = self.of.repro.create(container, path, duration=1)
+                result = False
+            else:
+                self.logger.info("launching repro: %s", job.config.name)
+                (container, path) = report
+                repro = self.of.repro.create(container, path, duration=1)
+                repros[job.job_id] = (job, repro)
 
-    def check_repro_impl(
-        self,
-    ) -> Tuple[bool, str, bool]:
-        # check all of the launched repros
+        return (result, repros)
 
-        self.cleared = False
+    def check_repro(self, repros: Dict[UUID, Tuple[Job, Repro]]) -> bool:
+        self.logger.info("checking repros")
+        self.success = True
 
-        def clear() -> None:
-            if not self.cleared:
-                self.cleared = True
-                print("")
+        def check_repro_impl() -> Tuple[bool, str, bool]:
+            # check all of the launched repros
 
-        commands: Dict[OS, Tuple[str, str]] = {
-            OS.windows: ("r rip", r"^rip=[a-f0-9]{16}"),
-            OS.linux: ("info reg rip", r"^rip\s+0x[a-f0-9]+\s+0x[a-f0-9]+"),
-        }
+            self.cleared = False
 
-        info: Dict[str, List[str]] = {}
+            def clear() -> None:
+                if not self.cleared:
+                    self.cleared = True
+                    print("")
 
-        done: Set[UUID] = set()
-        for job_id, repro in self.repros.items():
-            repro = self.of.repro.get(repro.vm_id)
-            if repro.error:
-                clear()
-                self.logger.error(
-                    "repro failed: %s: %s", self.target_jobs[job_id], repro.error
-                )
-                self.failed_jobs.add(job_id)
-                done.add(job_id)
-            elif repro.state not in [VmState.init, VmState.extensions_launch]:
-                done.add(job_id)
-                try:
-                    result = self.of.repro.connect(
-                        repro.vm_id,
-                        delete_after_use=True,
-                        debug_command=commands[repro.os][0],
+            commands: Dict[OS, Tuple[str, str]] = {
+                OS.windows: ("r rip", r"^rip=[a-f0-9]{16}"),
+                OS.linux: ("info reg rip", r"^rip\s+0x[a-f0-9]+\s+0x[a-f0-9]+"),
+            }
+
+            for (job, repro) in list(repros.values()):
```

Comment:
Is the `list()` here (and below) redundant, since we're turning an iterable into a list, which we then iterate over?

Reply:
This allows us to modify `repros` within the loop, which is otherwise not allowed.